wandb/sdk/launch/__init__.py

```diff
@@ -55,24 +59,42 @@ def _run(
     Helper that delegates to the project-running method corresponding to the passed-in backend.
     Returns a ``SubmittedRun`` corresponding to the project run.
     """
-    project = fetch_and_validate_project(
-        uri, experiment_name, api, runner_name, version, entry_point, parameters
-    )
 
+    src_project = "Uncategorized"
+    if _is_wandb_uri(uri):
+        src_project, _, _ = parse_wandb_uri(uri)
     if wandb_project is None:
-        wandb_project = api.settings("project") or "Uncategorized"
+        wandb_project = (
+            launch_config.get("project") or api.settings("project") or src_project
+        )
     if wandb_entity is None:
-        wandb_entity = api.settings("entity")
+        wandb_entity = launch_config.get("entity") or api.settings("entity")
+
+    experiment_name = experiment_name or launch_config.get("name")
+    project = fetch_and_validate_project(
+        uri,
+        wandb_entity,
+        wandb_project,
+        experiment_name,
+        api,
+        version,
+        entry_point,
+        parameters,
+    )
+    overrides = launch_config.get("overrides")
+    if overrides:
+        args = overrides.get("args")
+        if args:
+            args = _collect_args(args)
+            project._merge_parameters(args)
```

Comment:
so this just pulls everything into the project spec right

Reply:
yeah it collects the args into a dict, then merges it with the ones existing in the project, giving priority to existing parameters in the project (which are added via command line args)